File: versioned_docs/version-1.18.x/examples/py/src/data_replication_mode.py
import asyncio

from reduct import Client, ReplicationMode


async def main():
    # Update replication mode without altering other settings
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        # Pause to queue transactions without sending them
        await client.set_replication_mode("my_replication", ReplicationMode.PAUSED)

        # Disable to ignore new transactions
        await client.set_replication_mode("my_replication", ReplicationMode.DISABLED)

        # Re-enable to resume sending queued data
        await client.set_replication_mode("my_replication", ReplicationMode.ENABLED)


asyncio.run(main())

File: versioned_docs/version-1.18.x/examples/js/src/data_replication_mode.mjs
// reduct-js is published as CommonJS; in Node ESM examples we import the default export.
import reductJs from "reduct-js";

const { Client } = reductJs;

// Update replication mode without changing other settings
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });

// Pause to queue transactions without sending them
await client.setReplicationMode("my_replication", "paused");

// Disable to ignore new transactions
await client.setReplicationMode("my_replication", "disabled");

// Re-enable to resume sending queued data
await client.setReplicationMode("my_replication", "enabled");
